amazon/ion/writer_text.py

```diff
@@ -188,7 +209,10 @@ def _bytes_datetime(dt):
     if fractional_seconds == 0:
         tz_string += '.' + ('0' * fractional_precision)
     elif fractional_seconds is not None:
-        tz_string += str(fractional_seconds)[1:]
+        if 'E' in str(fractional_seconds):
```

Comment:
As far as I recall, `Decimal` can be represented as a full-precision decimal or via scientific notation with this [format](https://python-reference.readthedocs.io/en/latest/docs/float/scientific.html). 
And when it comes to casting as string, the Decimal as-is (whether it's represented as full precision decimal like 0.01 or scientific notation like 1E-9) should be retained when cast to string. 
For example, str(1E-9) == '1E-9'
I have accounted for the case where we see 'e' instead of 'E' for the scientific notation representation in the next commit.